src/components/fields/ArrayField.js

```diff
@@ -93,6 +100,78 @@ class ArrayField extends Component {
     const {items} = this.state;
     const {fields, definitions} = this.props.registry;
     const {SchemaField} = fields;
+    if (isFixedItems(schema)) {
```

Comment:
This render method is getting huge. Could you please extract this into its own component? You can define it in the same file, as a non exported symbol.